lib/kaguya/core.ex

```diff
@@ -49,21 +50,31 @@ defmodule Kaguya.Core do
   end
 
   def handle_info({:tcp, _socket, messages}, state) do
+    state = server_timer(state, server_timeout())
     for msg <- String.split(String.rstrip(messages), "\r\n"), do: handle_message(msg)
     {:noreply, state}
   end
 
   def handle_info({:ssl, _socket, messages}, state) do
+    state = server_timer(state, server_timeout())
     for msg <- String.split(String.rstrip(messages), "\r\n"), do: handle_message(msg)
     {:noreply, state}
   end
 
-  def handle_info({:tcp_closed, _port}, _state) do
+  def handle_info({:tcp_closed, _port}, state) do
+    cancel_server_timer(state)
     socket = reconnect()
     {:noreply, %{socket: socket}}
```

Comment:
Hm... now i'm thinking about it; I noticed that `:init` message isn't sent after reconnect also.... I was thinking of moving it there, but it seems Kaguya isn't going to init again after initial initialization.
(when handling :init Kaguya sends password/nick/username)